resources/aws/route_table.go

```diff
@@ -149,6 +155,32 @@ func (r RouteTable) MakePublic() error {
 	return nil
 }
 
+// CreateNatGatewayRoute creates a default route to the NAT gateway for the
+// private subnet. Retry is needed due to a delay while the gateway is created.
+func (r RouteTable) CreateNatGatewayRoute(natGatewayID string) (bool, error) {
+	createOperation := func() error {
+		_, err := r.Client.CreateRoute(&ec2.CreateRouteInput{
+			RouteTableId:         aws.String(r.id),
+			DestinationCidrBlock: aws.String("0.0.0.0/0"),
+			NatGatewayId:         aws.String(natGatewayID),
+		})
+		if err == nil || awsutil.IsRouteDuplicateError(err) {
+			return nil
+		}
+		if err != nil {
```

Comment:
nitpick, i think at this point `err` will be != from nil, the opposite of `err == nil || awsutil.IsRouteDuplicateError(err)` is `err != nil && !awsutil.IsRouteDuplicateError(err)`

Reply:
Good catch. I've cleaned up the error check so it falls through if the route already exists.